src/main/java/uk/gov/companieshouse/service/dissolution/DissolutionPatcher.java

```diff
@@ -107,6 +116,24 @@ private void addDirectorApproval(String userId, DissolutionPatchRequest body, Di
         director.setDirectorApproval(approval);
     }
 
+    private void updateSignatory(DissolutionPatchDirectorRequest body, Dissolution dissolution, String officerId) throws DissolutionNotFoundException, DirectorNotFoundException {
+        DissolutionDirector director = this.findDirector(officerId, dissolution);
+
+        if (director == null) {
+            throw new DirectorNotFoundException();
+        }
+
+        if (!director.getEmail().equals(body.getEmail()) || !Objects.equals(director.getOnBehalfName(), body.getOnBehalfName())) {
```

Comment:
Why Objects.equals? I'd also put this in a descriptive private method i.e. `hasDirectorDetailsChanged`

Reply:
Either of body.onBehalfName and director.onBehalfName can be null, so if we use getOnBehalfName().equals we risk a NPE. Using Objects.equals() we can check if both of these are the same values, and also check if both of them are null with no risk for NPE.

Reply:
ok we could probs use Java 8 Optional.ofNullable but whatever is cleaner. I'd still recommend the private method